### PR TITLE
docs: add storage

### PR DESCRIPTION
**Pull request by jiacai2050**

### CODE REVIEW

File: docs/src/storage.md [hunk 1]
```diff
@@ -1 +1,73 @@
...
+2. Under the premise of ensuring the correctness of the data, organize the data in the most reasonable way to meet the query needs of different scenarios.
+
+This document will introduce the internal implementation of the storage engine in CeresDB. Readers can refer to the content here to explore how to use CeresDB efficiently.
+
+
+# Overall structure
+
+CeresDB is a distributed storage system based on the share-nothing architecture. Data between different servers is isolated from each other and does not affect each other.The storage engine in each stand-alone machine is a variant of LSM (Log-structured merge-tree), which is optimized for time-series scenarios. The following figure shows its workflow.：
+
+TODO: Flow Chart
```
Comment (jiacai2050) [inline docs/src/storage.md:13]: ss
Will add this image when resources dir is created later.

File: docs/src/en/storage.md [hunk 2]
```diff
@@ -1 +1,67 @@
...
+
+This document will introduce the internal implementation of the storage engine in CeresDB. Readers can refer to the content here to explore how to use CeresDB efficiently.
+
+# Overall structure
+
+CeresDB is a distributed storage system based on the share-nothing architecture.
+
+Data between different servers is isolated from each other and does not affect each other. The storage engine in each stand-alone machine is a variant of [log-structured merge-tree](https://en.wikipedia.org/wiki/Log-structured_merge-tree), which is optimized for time-series scenarios. The following figure shows its core components:
+
+![](../resources/images/storage-overview.svg)
```
Comment (MachaelLee) [inline docs/src/en/storage.md:15]: I cannot see picture in rich diff mode.

File: docs/src/en/storage.md [hunk 3]
```diff
@@ -1 +1,67 @@
...
+
+In addition to storing the original data, the statistical information of the data will also be stored in the SST to speed up the query, such as the maximum value, the minimum value, etc.
+
+## Compactor
+
+Compactor can merge multiple small SST files into one, which is used to solve the problem of too many small files. In addition, Compactor will also delete expired data and duplicate data during the compaction. In future, compaction maybe add more task, such as downsample.
+
+The current compaction strategy in CeresDB reference Cassandra:
+
+- SizeTieredCompactionStrategy
```
Comment (MachaelLee) [inline docs/src/en/storage.md:52]: add a link of SizeTieredCompactionStrategy?

File: docs/src/en/storage.md [hunk 4]
```diff
@@ -1 +1,67 @@
...
+
+![](../resources/images/storage-overview.svg)
+
+## Write Ahead Log (WAL)
+
+A write request will be written to
+1. memtable in memory
+2. WAL in durable storage
+
+Since memtable is not persisted to the underlying storage system in real time, so WAL is required to ensure the reliability of the data in memtable.
```
Comment (MachaelLee) [inline docs/src/en/storage.md:23]: Add a link to WAL doc will be better, user can click on the link to learn more knowledge about WAL.

File: docs/src/en/storage.md [hunk 5]
```diff
@@ -1 +1,67 @@
...
+
+On the other hand, due to the design of the distributed architecture, WAL itself is required to be highly available. Now there are following implementations in CeresDB:
+-  Local disk (based on [RocksDB](http://rocksdb.org/), no distributed high availability)
+-  OceanBase
```
[the remark this answers is not in the document]
Comment (jiacai2050) [reply docs/src/en/storage.md:27]: There is no product named OceanBase KV, table API is one of its supported API.

File: docs/src/en/storage.md (hunk 2 above)
Comment (jiacai2050) [reply docs/src/en/storage.md:15]: image

You have to click `review file` to see images.

File: docs/src/en/storage.md (hunk 4 above)
Comment (jiacai2050) [reply docs/src/en/storage.md:23]: Yeah, I will add links in next PR after all current pending PR merged, the directory may also need refactor.

**Review by MachaelLee — APPROVED** [review]
LGTM

**Pull request merged**